src/CattyTests/Extensions/NSStringExtensionTests.swift

```diff
@@ -0,0 +1,253 @@
+
+        let testString2 = NSString(string: "abc")
+        let correctOutput2 = "a9993e364706816aba3e25717850c26c9cd0d89d"
+
+        let testString3 = NSString(string: "abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq")
+        let correctOutput3 = "84983e441c3bd26ebaae4aa1f95129e5e54670f1"
+
+        let testString4 = NSString(string: "abcdefghbcdefghicdefghijdefghijkefghijklfghijklmghijklmnhijklmnoijklmnopjklmnopqklmnopqrlmnopqrsmnopqrstnopqrstu")
+        let correctOutput4 = "a49b2446a02c645bf419f995b67091253a04a259"
+
+        let testString5 = NSString(string: "hello world")
+        let correctOutput5 = "2aae6c35c94fcfb415dbe95f408b9ce91ee846ed"
+
+        XCTAssertEqual(testString1.sha1(), correctOutput1)
+        XCTAssertEqual(testString2.sha1(), correctOutput2)
+        XCTAssertEqual(testString3.sha1(), correctOutput3)
+        XCTAssertEqual(testString4.sha1(), correctOutput4)
+        XCTAssertEqual(testString5.sha1(), correctOutput5)
+    }
+
+    func testStringByEscapingHTMLEntities() {
+
+        let testString1 = NSString(string: "&amp;")
+        let correctOutput1 = "&"
+
+        let testString2 = NSString(string: "&quot;")
+        let correctOutput2 = "\""
+
+        let testString3 = NSString(string: "&#x27;")
+        let correctOutput3 = "'"
+
+        let testString4 = NSString(string: "&#x39;")
+        let correctOutput4 = "'"
+
+        let testString5 = NSString(string: "&#x92;")
+        let correctOutput5 = "'"
+
+        let testString6 = NSString(string: "&#x96;")
+        let correctOutput6 = "'"
+
+        let testString7 = NSString(string: "&gt;")
+        let correctOutput7 = ">"
+
+        let testString8 = NSString(string: "&lt;")
+        let correctOutput8 = "<"
+
+        let testString9 = NSString(string: "")
+        let correctOutput9 = ""
+
+        let testString10 = NSString(string: "abcxyz.abc/xy=&amp;qbc&quot;&#x27;&lt;")
+        let correctOutput10 = "abcxyz.abc/xy=&qbc\"'<"
+
+        XCTAssertEqual(testString1.stringByEscapingHTMLEntities(), correctOutput1)
+        XCTAssertEqual(testString2.stringByEscapingHTMLEntities(), correctOutput2)
+        XCTAssertEqual(testString3.stringByEscapingHTMLEntities(), correctOutput3)
+        XCTAssertEqual(testString4.stringByEscapingHTMLEntities(), correctOutput4)
+        XCTAssertEqual(testString5.stringByEscapingHTMLEntities(), correctOutput5)
+        XCTAssertEqual(testString6.stringByEscapingHTMLEntities(), correctOutput6)
+        XCTAssertEqual(testString7.stringByEscapingHTMLEntities(), correctOutput7)
+        XCTAssertEqual(testString8.stringByEscapingHTMLEntities(), correctOutput8)
+        XCTAssertEqual(testString9.stringByEscapingHTMLEntities(), correctOutput9)
+        XCTAssertEqual(testString10.stringByEscapingHTMLEntities(), correctOutput10)
+
+    }
+
+    func testStringByEscapingForXMLValues() {
+
+        let testString1 = NSString(string: "<")
+        let correctOutput1 = "&amp;lt;"
```

Comment:
`<` should be converted to `&lt;` (without `&amp;`). It seems that the old implementation was incorrect too. `&amp;` should only be the result of an actual ampersand character.